lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,435 @@
+# -*- coding: utf-8 -*-
+######################################################################
+#                  Copyright (c) 2016 Northrop Grumman.
+#                          All rights reserved.
+######################################################################
+
+"""
+Flow analysis datatypes.
+"""
+
+import logging
+import re
+import subprocess
+
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.tabular import Tabular
+from . import data
+
+log = logging.getLogger(__name__)
+
+
+def is_number(s):
+    try:
+        float(s)
+        return True
+    except ValueError:
+        return False
+
+
+class FCS(Binary):
+    """Class describing an FCS binary file"""
+    file_ext = "fcs"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary FCS file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary FCSfile (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """
+        Checking if the file is in FCS format. Should read FCS2.0, FCS3.0
+        and FCS3.1
+
+        For this to work, need to have install checkFCS.R via bioconda
+        conda install ig-checkflowtypes
+        """
+        try:
+            rscript = 'checkFCS.R'
+            fcs_check = subprocess.check_output([rscript, filename])
+            if re.search('TRUE', str(fcs_check)):
+                return True
+            else:
+                return False
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format("fcs", "fcs", FCS)
+
+
+class FlowFrame(Binary):
+    """R Object containing flowFrame saved with saveRDS"""
+    file_ext = 'flowframe'
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary RDS flowFrame file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary RDS flowFrame (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """
+        Checking if the file is a flowFrame R object.
+
+        For this to work, need to have install checkFlowframe.R via bioconda
+        conda install ig-checkflowtypes
+        """
```

Comment:
Is this also possible via pip? Then this could be added to requirements?

Reply:
I have now removed the need for this, to avoid adding dependencies. Thanks!